### ISSUE TITLE
[Bug]: rootMode: "upward" ignores local browserslist config

### ISSUE THREAD

**Issue by AprilArcus** [babel/babel#14873]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

See minimal repro at https://github.com/AprilArcus/rootmode-upward-with-browserslist-repro

### Current and expected behavior

When invoking Babel with `rootMode: "upward"`, local configs in `.babelrc.js` are respected, but the `browserslist` entry in `package.json` is not.

Expected behavior: `browserslist` should be read from the local `package.json` or `.browserslistrc`, and fall back to the root `package.json` or `.browserslistrc` if a local configuration is not found.

I'm reporting this error in @babel/cli, but I initially encountered it in `babel-loader` and `babel-jest`.

### Environment

```
  System:
    OS: Linux 5.17 Debian GNU/Linux rodete 12 (rodete)
  Binaries:
    Node: 18.6.0 - /usr/bin/node
    npm: 8.15.1 - /usr/bin/npm
```

**Comment by nicolo-ribaudo** [comment]
For reference: `.browserslistrc` is supposed to be file-relative and not root/project-relative: https://github.com/babel/rfcs/pull/2#issuecomment-619573888

**Comment by AprilArcus** [comment]
I don't think I understand the comment you linked. It seems to me that a monorepo with multiple NPM workspaces in `myRepo/apps/*` be able to target each app to a different set of supported browsers, and have that config live in a single location that can be read by e.g. both [Webpack](https://webpack.js.org/configuration/target/#target) and Babel.

**Comment by nicolo-ribaudo** [comment]
Yep, I agree with your expected behavior :+1: That was a comment from the browserslist author saying that different folders should be able to target different configs.
